fix swift primary object store test

DeepDiver1975 opened the pull request.

DeepDiver1975 committed:
Wait for socket to be open

DeepDiver1975 commented:
green - https://ci.owncloud.org/job/core-swift-primary-storage-to-fix-it/

DeepDiver1975 commented:
:+1:

MorrisJobke commented:
👍

DeepDiver1975 commented:
Oooops


.PHP Fatal error:  Uncaught Error: Call to a member function instanceOfStorage() on null in /ssd/jenkins/workspace/core-swift-primary-storage-to-fix-it@2/database/mysql/label/SLAVE/lib/private/Files/View.php:1983
18:13:52 Stack trace:
18:13:52 #0 /ssd/jenkins/workspace/core-swift-primary-storage-to-fix-it@2/database/mysql/label/SLAVE/lib/private/Files/View.php(2037): OC\Files\View->unlockPath('/share4', 1, false)
18:13:52 #1 /ssd/jenkins/workspace/core-swift-primary-storage-to-fix-it@2/database/mysql/label/SLAVE/lib/private/Files/View.php(1168): OC\Files\View->unlockFile('/share4', 1)
18:13:52 #2 /ssd/jenkins/workspace/core-swift-primary-storage-to-fix-it@2/database/mysql/label/SLAVE/lib/private/Files/View.php(523): OC\Files\View->basicOperation('file_exists', '/share4')
18:13:52 #3 /ssd/jenkins/workspace/core-swift-primary-storage-to-fix-it@2/database/mysql/label/SLAVE/apps/files_sharing/lib/SharedMount.php(136): OC\Files\View->file_exists('/share4')
18:13:52 #4 /ssd/jenkins/workspace/core-swift-primary-storage-to-fix-it@2/database/mysql/label/SLAVE/apps/files_sharing/lib/SharedM in

PVince81 commented:
@DeepDiver1975 test is missing from the result lists ?!
Github shows all green

DeepDiver1975 committed:
Fix call on null

DeepDiver1975 commented:
> Github shows all green

The job is not yet active on pull requests - will reactivate once this is merged

DeepDiver1975 commented on lib/private/Files/View.php:
```diff
@@ -1980,7 +1980,7 @@ private function unlockPath($path, $type, $lockMountPoint = false) {
 		if ($mount) {
 			$storage = $mount->getStorage();
-			if ($storage->instanceOfStorage('\OCP\Files\Storage\ILockingStorage')) {
+			if ($storage && $storage->instanceOfStorage('\OCP\Files\Storage\ILockingStorage')) {
```
@icewind1991 @PVince81 I'd like to discuss if this only hides a problem or if this approach is valid - no idea why we cannot get the storage here in at least one case ....

icewind1991 replied:
`$storage` can be null if the storage class doesn't exist or if the storage constructor or storage wrappers throws an exception.

Both cases should be logged already

DeepDiver1975 commented:
Failing test:

````
05:32:52 There was 1 failure:
05:32:52 
05:32:52 1) OCA\Files_Sharing\Tests\MountProviderTest::testExcludeShares
05:32:52 Failed asserting that actual size 1 matches expected size 2.
05:32:52 
05:32:52 /ssd/jenkins/workspace/core-swift-primary-storage-to-fix-it/database/mysql/label/SLAVE/apps/files_sharing/tests/MountProviderTest.php:126
05:32:52

PVince81 commented:
Weird shit happening when testing this locally:
```
PHP Fatal error:  Maximum function nesting level of '100' reached, aborting! in Unknown on line 0
PHP Stack trace:
PHP   1. {main}() /home/vincent/opt/bin/phpunit:0
PHP   2. PHPUnit_TextUI_Command::main($exit = *uninitialized*) /home/vincent/opt/bin/phpunit:520
PHP   3. PHPUnit_TextUI_Command->run($argv = *uninitialized*, $exit = *uninitialized*) phar:///home/vincent/opt/bin/phpunit/phpunit/TextUI/Command.php:109
PHP   4. PHPUnit_TextUI_TestRunner->doRun($suite = *uninitialized*, $arguments = *uninitialized*, $exit = *uninitialized*) phar:///home/vincent/opt/bin/phpunit/phpunit/TextUI/Command.php:158
PHP   5. PHPUnit_Framework_TestSuite->run($result = *uninitialized*) phar:///home/vincent/opt/bin/phpunit/phpunit/TextUI/TestRunner.php:432
PHP   6. PHPUnit_Framework_TestSuite->run($result = *uninitialized*) phar:///home/vincent/opt/bin/phpunit/phpunit/Framework/TestSuite.php:747
PHP   7. call_user_func:{phar:///home/vincent/opt/bin/phpunit/phpunit/Framework/TestSuite.php:697}(*uninitialized*) phar:///home/vincent/opt/bin/phpunit/phpunit/Framework/TestSuite.php:697
PHP   8. OCA\Files_Versions\Tests\VersioningTest::setUpBeforeClass() phar:///home/vincent/opt/bin/phpunit/phpunit/Framework/TestSuite.php:697
PHP   9. OCA\Files_Versions\Tests\VersioningTest::loginHelper($user = *uninitialized*, $create = *uninitialized*) /srv/www/htdocs/owncloud/apps/files_versions/tests/VersioningTest.php:63
PHP  10. OC\Server->getUserFolder($userId = *uninitialized*) /srv/www/htdocs/owncloud/apps/files_versions/tests/VersioningTest.php:857
PHP  11. OC\Files\Node\Root->getUserFolder($userId = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/Server.php:809
PHP  12. OC_Util::copySkeleton($userId = *uninitialized*, $userDirectory = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/Files/Node/Root.php:352
PHP  13. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:320
PHP  14. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  15. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  16. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  17. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  18. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  19. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  20. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  21. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  22. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  23. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  24. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  25. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  26. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  27. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  28. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  29. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  30. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  31. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  32. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  33. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  34. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  35. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  36. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  37. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  38. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  39. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  40. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  41. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  42. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  43. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  44. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  45. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  46. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  47. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  48. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  49. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  50. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  51. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  52. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  53. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  54. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  55. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  56. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  57. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  58. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  59. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  60. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  61. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  62. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  63. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  64. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  65. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  66. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  67. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  68. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  69. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  70. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  71. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  72. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  73. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  74. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  75. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  76. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  77. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  78. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  79. OC_Util::copyr($source = *uninitialized*, $target = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:339
PHP  80. OC\Files\Node\Folder->newFolder($path = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/util.php:338
PHP  81. OC\Files\View->mkdir($path = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/Files/Node/Folder.php:144
PHP  82. OC\Files\View->basicOperation($operation = *uninitialized*, $path = *uninitialized*, $hooks = *uninitialized*, $extraParam = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/Files/View.php:261
PHP  83. OC\Files\View->writeUpdate($storage = *uninitialized*, $internalPath = *uninitialized*, $time = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/Files/View.php:1133
PHP  84. OC\Files\Cache\Updater->update($path = *uninitialized*, $time = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/Files/View.php:313
PHP  85. OC\Files\Cache\Updater->correctParentStorageMtime($internalPath = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/Files/Cache/Updater.php:136
PHP  86. OC\Files\Storage\Wrapper\Availability->filemtime($path = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/Files/Cache/Updater.php:244
PHP  87. OC\Files\Storage\Wrapper\Availability->checkAvailability() /srv/www/htdocs/owncloud/lib/private/Files/Storage/Wrapper/Availability.php:249
PHP  88. OC\Files\Storage\Wrapper\Availability->isAvailable() /srv/www/htdocs/owncloud/lib/private/Files/Storage/Wrapper/Availability.php:74
PHP  89. OC\Files\Storage\Wrapper\Wrapper->getAvailability() /srv/www/htdocs/owncloud/lib/private/Files/Storage/Wrapper/Availability.php:63
PHP  90. OC\Files\Storage\Wrapper\Wrapper->getAvailability() /srv/www/htdocs/owncloud/lib/private/Files/Storage/Wrapper/Wrapper.php:517
PHP  91. OC\Files\Storage\Common->getAvailability() /srv/www/htdocs/owncloud/lib/private/Files/Storage/Wrapper/Wrapper.php:517
PHP  92. OC\Files\Cache\Storage->getAvailability() /srv/www/htdocs/owncloud/lib/private/Files/Storage/Common.php:693
PHP  93. OC\Files\Cache\Storage::getStorageById($storageId = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/Files/Cache/Storage.php:144
PHP  94. OC_DB::executeAudited($stmt = *uninitialized*, $parameters = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/Files/Cache/Storage.php:81
PHP  95. OC_DB::prepare($query = *uninitialized*, $limit = *uninitialized*, $offset = *uninitialized*, $isManipulation = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/db.php:133
PHP  96. OC\DB\Connection->prepare($statement = *uninitialized*, $limit = *uninitialized*, $offset = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/legacy/db.php:65
PHP  97. OC\Server->getSystemConfig() /srv/www/htdocs/owncloud/lib/private/DB/Connection.php:163
PHP  98. OC\ServerContainer->query($name = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/Server.php:889
PHP  99. OC\AppFramework\Utility\SimpleContainer->query($name = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/ServerContainer.php:87
PHP 100. OC\AppFramework\Utility\SimpleContainer->sanitizeName($name = *uninitialized*) /srv/www/htdocs/owncloud/lib/private/AppFramework/Utility/
```

and this repeats.

PVince81 commented:
When running the MountProviderTest.php locally it also fails as above

PVince81 commented:
Mounts with "local" home storage:
```
$mounts                          = (array[2]);
  $mounts[0]                     = (object|OCA\Files_Sharing\SharedMount[11]);
    $mounts[0]->storage          = (null);
    $mounts[0]->recipientView    = (object|OC\Files\View[5])+;
    $mounts[0]->user             = (string[5]) 'user1';
    $mounts[0]->share            = (object|Mock_IShare_63645a71[2])+;
    $mounts[0]->class            = (string[24]) '\OC\Files\Storage\Shared';
    $mounts[0]->storageId        = (null);
    $mounts[0]->arguments        = (array[3])+;
    $mounts[0]->mountPoint       = (string[20]) '/user1/files/share2/';
    $mounts[0]->mountOptions     = (array[0]);
    $mounts[0]->*OC\Files\Mount\MountPoint*loader = (object|Mock_IStorageFactory_3030eae8[2])+;
    $mounts[0]->*OC\Files\Mount\MountPoint*invalidStorage = (bool) '0';
  $mounts[1]                     = (object|OCA\Files_Sharing\SharedMount[11]);
    $mounts[1]->storage          = (null);
    $mounts[1]->recipientView    = (object|OC\Files\View[5])+;
    $mounts[1]->user             = (string[5]) 'user1';
    $mounts[1]->share            = (object|Mock_IShare_63645a71[2])+;
    $mounts[1]->class            = (string[24]) '\OC\Files\Storage\Shared';
    $mounts[1]->storageId        = (null);
    $mounts[1]->arguments        = (array[3])+;
    $mounts[1]->mountPoint       = (string[20]) '/user1/files/share4/';
    $mounts[1]->mountOptions     = (array[0]);
    $mounts[1]->*OC\Files\Mount\MountPoint*loader = (object|Mock_IStorageFactory_3030eae8[2])+;
    $mounts[1]->*OC\Files\Mount\MountPoint*invalidStorage = (bool) '0';
```

Mounts with "swift" home storage:
```
$mounts                          = (array[1]);
  $mounts[0]                     = (object|OCA\Files_Sharing\SharedMount[11]);
    $mounts[0]->storage          = (null);
    $mounts[0]->recipientView    = (object|OC\Files\View[5])+;
    $mounts[0]->user             = (string[5]) 'user1';
    $mounts[0]->share            = (object|Mock_IShare_d4d0ab1e[2])+;
    $mounts[0]->class            = (string[24]) '\OC\Files\Storage\Shared';
    $mounts[0]->storageId        = (null);
    $mounts[0]->arguments        = (array[3])+;
    $mounts[0]->mountPoint       = (string[20]) '/user1/files/share4/';
    $mounts[0]->mountOptions     = (array[0]);
    $mounts[0]->*OC\Files\Mount\MountPoint*loader = (object|Mock_IStorageFactory_19c4799d[2])+;
    $mounts[0]->*OC\Files\Mount\MountPoint*invalidStorage = (bool) '0';
```

Something :japanese_ogre: ate the first share

PVince81 commented:
Ok it's because :japanese_ogre: ate an exception which said "Your test is not allowed to access the database". Now trying to debug and see what is databasing and why.

PVince81 commented:
Bääh... not mockable. It creates its own view internally.
And not bypassable. I guess I'll just allow DB access then...

Commit:
Allow DB access for MountProviderTest

Makes unit tests pass when using object store, since their FS access is
actually oc_filecache DB access. It is currently not possible to mock
or bypass the logic from "SharedMount::verifyMountPoint()" triggered by
this test.

PVince81 commented:
Are you kidding me ?? I did disable Xdebug.
```
....................EEEEEEEEEEEEEEEEEEEEEEEEEEEEEE..........E 7564 / 8088 ( 93%)
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE..... 7625 / 8088 ( 94%)
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE 7686 / 8088 ( 95%)
EEEEEEEEEEEEE.....................EEEEEEEEEEEEEEEEEEEEEEEEEEE 7747 / 8088 ( 95%)
EEEE..............................................EEEEEEEEEEE 7808 / 8088 ( 96%)
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE 7869 / 8088 ( 97%)
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE 7930 / 8088 ( 98%)
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE 7991 / 8088 ( 98%)
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE 8052 / 8088 ( 99%)
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE                          8088 / 8088 (100%)

Time: 5.86 minutes, Memory: 334.00Mb

There were 483 errors:

1) OCA\Files_Sharing\Tests\SharedMountTest::testPermissionMovedGroupShare with data set #136 ('folder', 9, 17)
include(/srv/www/htdocs/owncloud/3rdparty/guzzle/http/Guzzle/Http/Exception/MultiTransferException.php): failed to open stream: Too many open file
s
```

Seems like my system doesn't like running the Swift tests any more. It used to...

@DeepDiver1975 can you rerun the tests in your env ? I pushed a fix for the one failing test you reported.

DeepDiver1975 commented:
WOW - THX a lot @PVince81 - jenkins is green!
https://ci.owncloud.org/job/core-swift-primary-storage-to-fix-it/

:+1:

DeepDiver1975 commented:
I will backport as needed to fix ci on stable branches
